### PR TITLE
Improved Polarion test run updates.

### PR DESCRIPTION
**Pull request by sunilkumarn417**
- Improved update of polarion test run. 
- Fix suite name issue, to retrieve only suite name and not the suite file path.

**Test Runs:** 
( https://polarion.engineering.redhat.com/polarion/#/project/CEPH/testruns?query=author.id%3Asunnagar )

- **PASSED:**  

https://polarion.engineering.redhat.com/polarion/#/project/CEPH/testrun?id=test%5Fonly%5Ftrial1%5F16%5F2%5F6%2D11%5Fel8cp%5Ftier1%5Fssl%5Fdashboard%5Fport%5Frhel%2D8%5F4%5F0%2Dx86%5F64%2Dreleased%5FAutomated%5FSmoke%5FRuns

http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-CIQF3Z

xUnit: /ceph/cephci-jenkins/cephci-run-CIQF3Z/tier1_ssl_dashboard_port.xml

- **FAILED:**

https://polarion.engineering.redhat.com/polarion/#/project/CEPH/testrun?id=test%5Fonly%5Ftrial2%5F16%5F2%5F6%2D11%5Fel8cp%5Ftier1%5Fssl%5Fdashboard%5Fport%5Frhel%2D8%5F4%5F0%2Dx86%5F64%2Dreleased%5FAutomated%5FSmoke%5FRuns

http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-ISAFL5

xUnit: /ceph/cephci-jenkins/cephci-run-ISAFL5/tier1_ssl_dashboard_port.xml

Signed-off-by: sunilkumarn417 <sunnagar@redhat.com>

### CODE REVIEW

File: run.py [hunk 1]
```diff
@@ -981,6 +976,22 @@ def fetch_test_details(var) -> dict:
...
+            "suite-name": suite_name,
+            "distro": distro,
+            "ceph-version": ceph_version,
+            "ceph-ansible-version": ceph_ansible_version,
+            "test_cases": tcs,
+            "base_url": base_url,
+            "container-registry": docker_registry,
+            "container-image": docker_image,
+            "container-tag": docker_tag,
+            "compose-id": compose_id,
```
[the remark this answers is not in the document]
Comment (sunilkumarn417) [reply run.py:991]: yes

File: utility/polarion.py [hunk 2]
```diff
@@ -88,3 +89,98 @@ def post_to_polarion(tc):
...
+            "Automated_Smoke_Runs",
+        ]
+    )
+    test_run["test_run_id"] = test_run["test_run_id"].replace(".", "_")
+
+    test_run["ceph-build"] = "_".join(
+        [
+            test_run["ceph-version"],
+            test_run["ceph-ansible-version"],
+            test_run["compose-id"],
```
[the remark this answers is not in the document]
Comment (sunilkumarn417) [reply utility/polarion.py:119]: Currently we still have to integrate cephadm-ansible into framework.
And ceph-ansible still required in 4x to 5x upgrade.

File: run.py [hunk 3]
```diff
@@ -29,7 +29,7 @@
     create_ceph_nodes,
     create_ibmc_ceph_nodes,
 )
-from utility.polarion import post_to_polarion
+from utility.polarion import post_to_polarion_v2
```
[the remark this answers is not in the document]
Comment (sunilkumarn417) [reply run.py:32]: sure

**Comment by veera-raghava-reddy** [comment]
@sunilkumarn417 Any blocker to proceed with this PR?

**Comment by sunilkumarn417** [comment]
> @sunilkumarn417 Any blocker to proceed with this PR?

Yes @veera-raghava-reddy , we need get alternative way to login into Polarion server, since only SSO-SAML authorization is supported now.

**Comment by veera-raghava-reddy** [comment]
@sunilkumarn417 POlarions SSO issue is resolved?

**Comment by sunilkumarn417** [comment]
> @sunilkumarn417 Polarion SSO issue is resolved?

Yes @veera-raghava-reddy , we have one solution now and using this solution, currently thinking about different workflows to update test results in polarion which needs to suit our current pipeline.

**Comment by sunilkumarn417** [comment]
closing this PR, as we had this issue addressed in already Merged PR(https://github.com/red-hat-storage/cephci/pull/1092).

**Pull request closed without merging**